test/cli/pinning_remote_test.go

```diff
@@ -0,0 +1,421 @@
+		t.Parallel()
+
+		verifyStatus := func(node *harness.Node, name, hash, status string) {
+			resJSON := node.IPFS("pin", "remote", "ls",
+				"--service=svc",
+				"--enc=json",
+				"--name="+name,
+				"--status="+status,
+			).Stdout.String()
+
+			assert.Equal(t, status, gjson.Get(resJSON, "Status").Str)
+			assert.Equal(t, hash, gjson.Get(resJSON, "Cid").Str)
+			assert.Equal(t, name, gjson.Get(resJSON, "Name").Str)
+		}
+
+		t.Run("'ipfs pin remote add --background=true'", func(t *testing.T) {
+			node := harness.NewT(t).NewNode().Init().StartDaemon()
+			svc, svcURL := runPinningService(t, authToken)
+			node.IPFS("pin", "remote", "service", "add", "svc", svcURL, authToken)
+
+			// retain a ptr to the pin that's in the DB so we can directly mutate its status
+			// to simulate async work
+			pinCh := make(chan *pinningservice.PinStatus, 1)
+			svc.PinAdded = func(req *pinningservice.AddPinRequest, pin *pinningservice.PinStatus) {
+				pinCh <- pin
+			}
+
+			hash := node.IPFSAddStr("foo")
+			node.IPFS("pin", "remote", "add",
+				"--background=true",
+				"--service=svc",
+				"--name=pin1",
+				hash,
+			)
+
+			pin := <-pinCh
+
+			transitionStatus := func(status string) {
+				pin.M.Lock()
+				pin.Status = status
+				pin.M.Unlock()
+			}
+
+			verifyStatus(node, "pin1", hash, "queued")
+
+			transitionStatus("pinning")
+			verifyStatus(node, "pin1", hash, "pinning")
+
+			transitionStatus("pinned")
+			verifyStatus(node, "pin1", hash, "pinned")
+
+			transitionStatus("failed")
+			verifyStatus(node, "pin1", hash, "failed")
+		})
+
+		t.Run("'ipfs pin remote add --background=false'", func(t *testing.T) {
+			t.Parallel()
+			node := harness.NewT(t).NewNode().Init().StartDaemon()
+			svc, svcURL := runPinningService(t, authToken)
+			node.IPFS("pin", "remote", "service", "add", "svc", svcURL, authToken)
+
+			svc.PinAdded = func(req *pinningservice.AddPinRequest, pin *pinningservice.PinStatus) {
+				pin.M.Lock()
+				defer pin.M.Unlock()
+				pin.Status = "pinned"
+			}
+			hash := node.IPFSAddStr("foo")
+			node.IPFS("pin", "remote", "add",
+				"--background=false",
+				"--service=svc",
+				"--name=pin2",
+				hash,
+			)
+			verifyStatus(node, "pin2", hash, "pinned")
+		})
+
+		t.Run("'ipfs pin remote ls' with multiple statuses", func(t *testing.T) {
+			t.Parallel()
+			node := harness.NewT(t).NewNode().Init().StartDaemon()
+			svc, svcURL := runPinningService(t, authToken)
+			node.IPFS("pin", "remote", "service", "add", "svc", svcURL, authToken)
+
+			hash := node.IPFSAddStr("foo")
+			desiredStatuses := map[string]string{
+				"pin-queued":  "queued",
+				"pin-pinning": "pinning",
+				"pin-pinned":  "pinned",
+				"pin-failed":  "failed",
+			}
+			svc.PinAdded = func(req *pinningservice.AddPinRequest, pin *pinningservice.PinStatus) {
+				pin.M.Lock()
+				defer pin.M.Unlock()
+				pin.Status = desiredStatuses[pin.Pin.Name]
+			}
+			// TODO this is racy, either dont' use backgorund=true or synchronize w/ channels
```

Comment:
Can you edit the comment to explain why this racy code is acceptable here please ?

It is not obvious to me.